application/modules/shop/actions/BatchEditPriceAction.php

```diff
@@ -0,0 +1,226 @@
+<?php
+namespace app\modules\shop\actions;
+
+use Yii;
+use yii\base\Action;
+use yii\web\Response;
+use app\modules\shop\models\Category;
+use app\modules\shop\models\Currency;
+use app\modules\shop\models\Product;
+
+class BatchEditPriceAction extends Action
+{
+    const BEP_CONTEXT_PRODUCT = 'backend-product';
+    const BEP_KIND_FIXED = 'fixed';
+    const BEP_TYPE_NORMAL = 'normal';
+    const BEP_FIELD_PRICE = 'price';
+    const BEP_FIELD_OLDPRICE = 'old_price';
+    const BEP_FIELD_ALL = 'all';
+    const BEP_INCREMENT = 'inc';
+
+    public $prm = [];
+    
+    public function run()
+    {
+        if (false === Yii::$app->request->isAjax) {
+            throw new NotFoundHttpException();
+        }
+
+        Yii::$app->response->format = Response::FORMAT_JSON;
+        Yii::$app->log->flushInterval = 100;
+        Yii::$app->log->traceLevel = 0;
+        Yii::$app->log->targets = [
+            new yii\log\FileTarget(
+                [
+                    'levels' => ['error'],
+                    'exportInterval' => 100
+                ]
+            )
+        ];
```

Comment:
That's kinda weird log magic - this can produce hell to developers who'd like to trace this stuff